tests/ops/qubit/test_parametric_ops.py

```diff
@@ -2780,13 +2783,17 @@ def test_PauliRot_all_Identity_broadcasted(self):
         op = qml.PauliRot(theta, "II", wires=[0, 1])
         decomp_ops = op.decomposition()
 
-        phases = np.exp(-1j * theta / 2)
-        assert np.allclose(op.eigvals(), np.outer(phases, np.ones(4)))
-        mat = op.matrix()
-        for phase, sub_mat in zip(phases, mat):
-            assert np.allclose(sub_mat, phase * np.eye(4))
+        assert len(decomp_ops) == 1
+
+        decomp_op = decomp_ops[0]
+
+        assert qml.equal(decomp_op, qml.GlobalPhase(theta / 2))
 
-        assert len(decomp_ops) == 0
+        op_matrices = op.matrix()
+        decomp_op_matrices = decomp_op.matrix().T
```

Comment:
Wait why the transpose? 

Also, as a note you can do `decomp_mat = qml.matrix(op.decomposition)()`.

Reply:
I added the transpose because the original shape of `op.decompose()[0].matrix()` is `(1,3)` so since I want to compare lengths and then iterate over each of the 3 elements, I transpose it.

Reply:
I tried using your suggestion but it complains about wires, and even after doing:
`qml.matrix(op.decomposition, wire_order=[0,1])()`

I get the error: MatrixUndefinedError: 

I think it doesn't work because decomposition would return a list (?)

Reply:
After investigating locally, I found bug #5880 .

As noted, we do need wire orders to actually compare the two matrices, but when we provide the wire order, we have issues. This failure in computing the matrix is why calling `qml.matrix(op.decomposition, wire_order=[0,1])()` is giving a `MatrixUndefinedError`.

Reply:
I see. So it's an issue with GlobalPhase particularly 😮

Reply:
Do you reckon I have answered the comments you left? @albi3ro 😄